src/fsrs_optimizer/fsrs_simulator.py

```diff
@@ -345,17 +418,32 @@ def mean_reversion(init, current):
         card_table[col["stability"]][reviewed] = new_s[reviewed]
         card_table[col["difficulty"]][reviewed] = new_d[reviewed]
 
-        card_table[col["ivl"]][reviewed] = np.clip(
-            next_interval(
-                card_table[col["stability"]][reviewed],
-                request_retention,
-                fuzz=fuzz,
-                decay=-w[20],
-            ),
-            1,
-            max_ivl,
-        )
-        card_table[col["due"]][reviewed] = today + card_table[col["ivl"]][reviewed]
+        if scheduler_name == "fsrs":
+            card_table[col["ivl"]][reviewed] = np.clip(
+                next_interval(
+                    card_table[col["stability"]][reviewed],
+                    request_retention,
+                    fuzz=fuzz,
+                    decay=-w[20],
+                ),
+                1,
+                max_ivl,
+            )
+            card_table[col["due"]][reviewed] = today + card_table[col["ivl"]][reviewed]
+        else:  # anki scheduler
+            reviewed_indices = np.where(reviewed)[0]
+            last_dates = card_table[col["last_date"]][reviewed_indices]
+            dues = card_table[col["due"]][reviewed_indices]
+            eases = card_table[col["ease"]][reviewed_indices]
+            ivls = card_table[col["delta_t"]][reviewed_indices]
+            ratings = card_table[col["rating"]][reviewed_indices].astype(int)
+
+            delta_ts, new_eases = anki_sm2_scheduler(
+                dues - last_dates, ivls, eases, ratings
+            )
+            card_table[col["ivl"]][reviewed_indices] = delta_ts
+            card_table[col["due"]][reviewed_indices] = today + delta_ts
+            card_table[col["ease"]][reviewed_indices] = new_eases
```

Comment:
reviewed_indices -> reviewed
```suggestion
            last_dates = card_table[col["last_date"]][reviewed]
            dues = card_table[col["due"]][reviewed]
            eases = card_table[col["ease"]][reviewed]
            ivls = card_table[col["delta_t"]][reviewed]
            ratings = card_table[col["rating"]][reviewed].astype(int)

            delta_ts, new_eases = anki_sm2_scheduler(
                dues - last_dates, ivls, eases, ratings
            )
            card_table[col["ivl"]][reviewed] = delta_ts
            card_table[col["due"]][reviewed] = today + delta_ts
            card_table[col["ease"]][reviewed] = new_eases
```

I can't see any difference?
![image](https://github.com/user-attachments/assets/5d7508bc-e6ca-46b8-ad81-a79432de97f1)